Clinical-Genomics/scout
# Fix GT ref and alt depth parsing when values are 0

When you have the following genotype situation for a SV or fusion variant in the VCF:

`GT:DV:RV:FFPM	./1:0:0:0.0`

Then the variants gets parsed like this:

`{'individual_id': '888-88', 'display_name': '888-88', 'genotype_call': './1', 'so': None, 'alt_depth': -1, 'ref_depth': -1, 'read_depth': -1, 'alt_frequency': -1.0, 'genotype_quality': -1, 'ffpm': 0, 'split_read': 0}`

But instead we'd expect the alt_depth and read_depth to be 0

This is a fix (fix #4272)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>


**How to test**:
1. Run scout setup demo on main branch and go to fusion variantS page, the last variant should have -1 as junction Reads

<img width="597" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/d8bbe267-dd60-4cfc-afb0-03767baf3edf">

1. repeat with this branch and notice that the junction reads are 0

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
